src/implementations/vanilla/components/news-ticker/news-ticker.js

```diff
@@ -0,0 +1,275 @@
+    return newsTicker;
+  }
+
+  constructor(
+    element,
+    {
+      toggleSelector = '[data-ecl-news-ticker-toggle]',
+      prevSelector = '[data-ecl-news-ticker-prev]',
+      nextSelector = '[data-ecl-news-ticker-next]',
+      containerClass = '.ecl-news-ticker__container',
+      contentClass = '.ecl-news-ticker__content',
+      slidesClass = '.ecl-news-ticker__slides',
+      slideClass = '.ecl-news-ticker__slide',
+      currentSlideClass = '.ecl-news-ticker__counter--current',
+      attachClickListener = true,
+      attachResizeListener = true,
+    } = {}
+  ) {
+    // Check element
+    if (!element || element.nodeType !== Node.ELEMENT_NODE) {
+      throw new TypeError(
+        'DOM element should be given to initialize this widget.'
+      );
+    }
+
+    this.element = element;
+
+    // Options
+    this.toggleSelector = toggleSelector;
+    this.prevSelector = prevSelector;
+    this.nextSelector = nextSelector;
+    this.containerClass = containerClass;
+    this.contentClass = contentClass;
+    this.slidesClass = slidesClass;
+    this.slideClass = slideClass;
+    this.currentSlideClass = currentSlideClass;
+    this.attachClickListener = attachClickListener;
+    this.attachResizeListener = attachResizeListener;
+
+    // Private variables
+    this.toggle = null;
+    this.container = null;
+    this.content = null;
+    this.slides = null;
+    this.btnPrev = null;
+    this.btnNext = null;
+    this.index = 1;
+    this.total = 0;
+    this.allowShift = true;
+    this.autoPlay = false;
+    this.autoPlayInterval = null;
+    this.resizeTimer = null;
+
+    // Bind `this` for use in callbacks
+    this.handleClickOnToggle = this.handleClickOnToggle.bind(this);
+    this.shiftSlide = this.shiftSlide.bind(this);
+    this.checkIndex = this.checkIndex.bind(this);
+    this.moveSlides = this.moveSlides.bind(this);
+    this.resizeTicker = this.resizeTicker.bind(this);
+    this.handleResize = this.handleResize.bind(this);
+  }
+
+  /**
+   * Initialise component.
+   */
+  init() {
+    this.toggle = queryOne(this.toggleSelector, this.element);
+    this.btnPrev = queryOne(this.prevSelector, this.element);
+    this.btnNext = queryOne(this.nextSelector, this.element);
+    this.slidesContainer = queryOne(this.slidesClass, this.element);
+    this.container = queryOne(this.containerClass, this.element);
+    this.content = queryOne(this.contentClass, this.element);
+
+    const slidesOrigin = queryAll(this.slideClass, this.element);
+    this.total = slidesOrigin.length;
+
+    const firstSlide = slidesOrigin[0];
+    const lastSlide = slidesOrigin[slidesOrigin.length - 1];
+    const cloneFirst = firstSlide.cloneNode(true);
+    const cloneLast = lastSlide.cloneNode(true);
+
+    // Clone first and last slide
+    this.slidesContainer.appendChild(cloneFirst);
+    this.slidesContainer.insertBefore(cloneLast, firstSlide);
+
+    this.slides = queryAll(this.slideClass, this.element);
```

Comment:
didn't we make this query already in `SlideOrigin`?

Reply:
I made a small update and added comments, i need to refresh the slides variable after adding new cloned slides.